tidy3d/components/simulation.py

```diff
@@ -437,6 +437,48 @@ def _warn_monitor_simulation_frequency_range(cls, val, values):
                 )
         return val
 
+    @pydantic.validator("monitors", always=True)
+    def diffraction_efficiency_size(cls, val, values):
+        """If any :class:`DiffractionEfficiencyMonitor` exists, ensure that its size is at least
+        as large as the simulation domain in the transverse direction."""
+        monitors = val
+        sim_size = values.get("size")
+        for monitor in monitors:
+            if not isinstance(monitor, DiffractionEfficiencyMonitor):
```

Comment:
let's just do

```py
if isinstance(monitor, DiffractionEfficiencyMonitor):
    logic
```

instead of 
```py
if not isinstance(monitor, DiffractionEfficiencyMonitor):
    continue
logic
```

Reply:
same for the validator below, it's very minor but I think just makes it easier to follow and extend later if we add more monitors.